### PR TITLE
Make http headers case insensitive

### PR DESCRIPTION
**Pull request by arukshani**
Move header operations to mime package since those can be reused for multiparts as well

## Automation tests
 - Unit tests 
   > Added

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

```ballerina
import ballerina.net.http;
import ballerina.mime;
import ballerina.io;

@http:configuration {port:9093}
service<http> test {
    @http:resourceConfig {
        methods:["POST"],
        path:"/testHeader"
    }
    resource headerTest (http:Connection conn, http:InRequest inRequest) {
        io:println(inRequest.getHeader("ConTeNT-Type"));
        http:OutRequest outRequest = {};
        outRequest.setHeader("Test-HEADer", "Hello");
        io:println(outRequest.getHeader("test-headeR"));

        http:OutResponse res = {};
        _ = conn.respond(res);
    }
}
```

### CODE REVIEW

File: tests/ballerina-test/src/test/java/org/ballerinalang/test/mime/HeaderTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
+package org.ballerinalang.test.mime;
```
Comment (djkevincr) [inline tests/ballerina-test/src/test/java/org/ballerinalang/test/mime/HeaderTest.java]: Shall we add license header?

File: tests/ballerina-test/src/test/java/org/ballerinalang/test/mime/HeaderTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import org.ballerinalang.launcher.util.BCompileUtil;
+import org.ballerinalang.launcher.util.BRunUtil;
+import org.ballerinalang.launcher.util.CompileResult;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BValue;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class HeaderTest {
```
Comment (djkevincr) [inline tests/ballerina-test/src/test/java/org/ballerinalang/test/mime/HeaderTest.java]: Shall we add class level java doc?

File: tests/ballerina-test/src/test/java/org/ballerinalang/test/mime/HeaderTest.java (hunk 1 above)
Comment (arukshani) [reply tests/ballerina-test/src/test/java/org/ballerinalang/test/mime/HeaderTest.java]: done

File: stdlib/ballerina-mime/src/main/ballerina/ballerina/mime/natives.bal [hunk 3]
```diff
@@ -277,3 +277,141 @@ public const string DEFAULT_CHARSET = "UTF-8";
...
+public function <Entity entity> addHeader (string headerName, string headerValue) {
+    if (entity.headers == null) {
+        entity.headers = {};
+    }
+    string[] existingValues = useCaseInsensitiveStrategy(headerName, entity.headers);
+    if (existingValues == null) {
+        entity.setHeader(headerName, headerValue);
+    } else {
+        existingValues[lengthof existingValues] = headerValue;
+        string existingHeaderName = getCaseInsensitiveHeaderName(headerName, entity.headers);
```
Comment (pubudu91) [inline stdlib/ballerina-mime/src/main/ballerina/ballerina/mime/natives.bal]: What's the need for looking up an existing header name here? If we do that, basically we are changing the header name the user set isn't it?

Comment (arukshani) [reply stdlib/ballerina-mime/src/main/ballerina/ballerina/mime/natives.bal]: if a header exists regardless of its case, we add the value to the same header. User can't change the header name if it exists already unless he does a setHeader

**Comment by arukshani** [comment]
New PR >> https://github.com/ballerina-lang/ballerina/pull/5339

**Pull request closed without merging**